public/tier0/threadtools.h

```diff
@@ -664,35 +701,41 @@ class PLATFORM_CLASS CThreadMutex
 
 #if !defined(THREAD_PROFILER)
 
-class CThreadFastMutex
+class CThreadSpinMutex
 {
 public:
-	CThreadFastMutex()
+	CThreadSpinMutex( const char* pDebugName = NULL )
 	  :	m_ownerID( 0 ),
-	  	m_depth( 0 )
+	  	m_depth( 0 ),
+		m_pDebugName( NULL/*pDebugName*/ )
 	{
 	}
-
+	
 private:
-	FORCEINLINE bool TryLockInline( const uint32 threadId ) volatile
+	FORCEINLINE bool TryLockInline( const char *pFileName, int nLine, const ThreadId_t threadId ) volatile
 	{
-		if ( threadId != m_ownerID && !ThreadInterlockedAssignIf( (volatile int32 *)&m_ownerID, (int32)threadId, 0 ) )
+		if ( threadId != m_ownerID && ( m_ownerID ||
+#ifdef _WIN32

```

Comment:
This would only work on win, I guess it's better to check for PLATFORM_64BITS in here instead?

Reply:
Okay, nvm, I see why it's like that